pin/pin.go

```diff
@@ -213,11 +213,6 @@ func NewPinner(dstore ds.Datastore, serv, internal ipld.DAGService) Pinner {
 func (p *pinner) Pin(ctx context.Context, node ipld.Node, recurse bool) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	err := p.dserv.Add(ctx, node)
```

Comment:
I believe we still need to _add_ this node, we can probably just do it outside the lock.

Reply:
I basically reverted [this commit](https://github.com/ipfs/go-ipfs/commit/953b610470044bd332522b346e4881b8957c2050#diff-10c79301e027d970f7cba49f4a853d86), ie I'm assuming that we have already added before `Pin()` is called.
Is that incorrect?

Reply:
I really don't know. The safe solution would be to change the API. Let's pause this for a few days while I try to pull in a CoreAPI patch, then we can look into making `Pin` take a CID.

Reply:
This is done outside the lock now:
https://github.com/ipfs/go-ipfs-pinner/blob/master/dspinner/pin.go#L177